### PR TITLE
Move amp-carousel tests away from the iframe framework

### PR DESCRIPTION
**Pull request by nainar**
All tests pass now. 

Stale:
<del>Currently skipping one test. But thought it better to commit all the working tests first and debug the last non working test and commit the working version in a separate test. </del>

<del>Skipped test title: `(type=slides loop) should always have a prev and next button be able to get past the first and last item`</del>

### CODE REVIEW

File: test/integration/test-amp-carousel.js [hunk 1]
```diff
@@ -14,249 +14,351 @@
...
+    });
+
+    it('should be present', () => {
+      expect(document.querySelectorAll('amp-carousel')).to.have.length.above(0);
+    });
+
+    it('should not have the buttons visible ' +
+        'when amp-mode-mouse class is not on body', () => {
+      document.body.classList.remove('amp-mode-mouse');
+      return Promise.resolve().then(timeout(2000))
```
Comment (cathyxz) [inline test/integration/test-amp-carousel.js]: Can we do better than "wait for 2 seconds" here? What is it actually waiting for?

File: test/integration/test-amp-carousel.js [hunk 2]
```diff
@@ -14,249 +14,351 @@
...
+          });
+    });
 
-  it('should be present', () => {
-    expect(fixture.doc.querySelectorAll('amp-carousel'))
-        .to.have.length.above(0);
-    return fixture.awaitEvent(AmpEvents.LOAD_START, 1).then(() => {
-      expect(fixture.doc.querySelectorAll('amp-carousel'))
-          .to.have.length.above(0);
+    it('should not have any buttons enabled when theres only a single ' +
```
Comment (cathyxz) [inline test/integration/test-amp-carousel.js]: Is there a difference here between visible and enabled?

File: test/integration/test-amp-carousel.js [hunk 3]
```diff
@@ -14,249 +14,351 @@
...
+            const nextBtn = amp.querySelector('.amp-carousel-button-next');
+            expect(document.body).to.not.have.class('amp-mode-mouse');
+            expect(prevBtn).to.not.be.null;
+            expect(nextBtn).to.not.be.null;
+            expect(prevBtn).to.be.hidden;
+            expect(nextBtn).to.be.hidden;
+          });
+    });
+
+    it('should have the buttons visible when amp-mode-mouse ' +
```
Comment (cathyxz) [inline test/integration/test-amp-carousel.js]: Only the next button visible, right?

File: test/integration/test-amp-carousel.js [hunk 4]
```diff
@@ -14,249 +14,351 @@
...
+            const prevBtn = amp.querySelector('.amp-carousel-button-prev');
+            const nextBtn = amp.querySelector('.amp-carousel-button-next');
+            expect(prevBtn).to.not.be.null;
+            expect(nextBtn).to.not.be.null;
+            expect(prevBtn).to.be.hidden;
+            expect(nextBtn).to.be.visible;
+          });
     });
 
     it('should not be able to go past the first or last item', () => {
```
Comment (cathyxz) [inline test/integration/test-amp-carousel.js]: Do you have a test case where both buttons are visible?

File: test/integration/test-amp-carousel.js [hunk 5]
```diff
@@ -14,249 +14,351 @@
...
-
-        expect(amp).to.have.attribute('controls');
-        expect(amp).to.have.class('i-amphtml-carousel-has-controls');
-
-        const prevBtn = amp.querySelector('.amp-carousel-button-prev');
-        const nextBtn = amp.querySelector('.amp-carousel-button-next');
-        expect(prevBtn).to.be.hidden;
-        expect(nextBtn).to.be.visible;
-      });
+    it.skip('(type=slides loop) should always have a prev and next button be ' +
```
Comment (cathyxz) [inline test/integration/test-amp-carousel.js]: Intentionally skipped?

**Comment by nainar** [comment]
@cathyxz uploaded the latest PR with the passing test.

**Comment by cathyxz** [comment]
Awesome. LGTM.

**Pull request merged**